real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/sensors/LoggingNode.java

```diff
@@ -154,7 +156,9 @@ private void setupSubscriberList() {
             new Subscriber("log", filter.getMsgPath(), new MessageListener() {
                 @Override
                 public void actionPerformed(String topicName, Message m) {
-                    messageQueue.add(m);
+                    while (!messageQueue.offer(m)){
```

Comment:
we may want to rethink this, since this has the potential to turn into an infinite loop. I vote if we can't offer it to the queue, drop the oldest message